python/lsst/analysis/drp/gatherResourceUsage.py

```diff
@@ -61,6 +61,107 @@
 _LOG = logging.getLogger(__name__)
 
 
+class ConsolidateResourceUsageConnections(PipelineTaskConnections, dimensions=()):
+    output_table = ct.Output(
+        name="ResourceUsageSummary",
+        storageClass="DataFrame",
+        dimensions=(),
+        doc="Consolidated table of resource usage statistics. One row per task label",
+    )
+
+    def __init__(self, *, config):
+        super().__init__(config=config)
+        for name in self.config.input_names:
+            setattr(
+                self,
+                name,
+                ct.Input(
+                    name,
+                    storageClass="DataFrame",
+                    dimensions=(),
+                    doc="Resource usage statistics for a task.",
+                ),
+            )
+            self.inputs.add(name)
+
+
+class ConsolidateResourceUsageConfig(
+    PipelineTaskConfig, pipelineConnections=ConsolidateResourceUsageConnections
+):
+    input_names = ListField(
+        doc="Input resource usage dataset type names",
+        dtype=str,
+        default=[],
+    )
+
+
+class ConsolidateResourceUsageTask(PipelineTask):
+    """A `PipelineTask` that summarizes task resource usage into a single
+    table with per-task rows.
+
+    Notes
+    -----
+    This is an unusual `PipelineTask` in that its input connection has
+    dynamic dimensions, and its quanta are generally built via a custom
+    quantum-graph builder defined in the same module.
+    """
+
+    ConfigClass = ConsolidateResourceUsageConfig
+    _DefaultName = "consolidateResourceUsage"
+
+    def run(self, **kwargs):
+        quantiles = []
+        for input_name, ru_table in kwargs.items():
+            if not input_name.endswith("resource_usage"):
+                continue
+            else:
+                df = ru_table.quantile(
+                    [0.0, 0.01, 0.05, 0.32, 0.50, 0.68, 0.95, 0.99, 1.0],
+                    numeric_only=True,
+                ).reset_index()
+                df["task"] = input_name.replace("_resource_usage", "")
+                df["quanta"] = len(ru_table)
+                df["integrated_runtime"] = ru_table["run_time"].sum()
+
+                quantiles.append(
+                    df[
+                        [
+                            "index",
+                            "quanta",
+                            "task",
+                            "memory",
+                            "init_time",
```

Comment:
I see that we store `init_time` and `run_time`. Would it be valuable to have a human-readable stop-time also? I'm not sure how useful these quantities are for identifying transient infrastructure errors.

Reply:
These are actually both durations, not timestamps: `init_time` is the time it takes to do all of the stuff `pipetask` does before it actually call's the task's `runQuantum` method, which amounts to a lot of file-existence checks most of the time. That's the time when things like "Nothing to do for task" or "Skipping already-successful quantum" happen.